### PR TITLE
Extend Multi check for bandmap display

### PR DESCRIPTION
**Pull request by dl1jbe**
Supports all builtin contests besides WPX, PACC_PA and FOC.

* Evaluates worked zones and countries 
* Honor DX_&_SECTIONS, COUNTRY_MULT, ITUMULT and WAZMULT keyword for other contests.

Multi based on exchange field (e.g. worked sections) can not be checked for bandmap display. 

Do not check in the code yet. I will add WPX and PACC shortly (next days).
Please test and give feedback.

### CODE REVIEW

**Comment by dl1jbe** [comment]
Btw, code needs quite some refactoring, which can/should be done after test.

File: src/bandmap.c [hunk 1]
```diff
@@ -447,20 +447,17 @@ void bandmap_age() {
  *
  * \return true if new multi
  */
-bool bm_ismulti(char *call, spot *data, int band) {
+bool bm_ismulti(spot *data, int band) {
```
Comment (zcsahok) [inline src/bandmap.c:450]: maybe even `band` is not needed if it's always called with `data->band`.

File: src/setcontest.c [hunk 2]
```diff
@@ -26,13 +26,65 @@
...
+    }
+
+    if (country_mult == 1) {
+	return ((countries[data->ctynr] & inxes[band]) == 0);
+    }
+
+    /* TODO: pfxmultab */
+
+    if ((itumult == 1) || (wazmult == 1)) {
+	return ((zones[data->cqzone] && inxes[band]) == 0);
```
Comment (zcsahok) [inline src/setcontest.c:80]: typo? && -> &

File: src/setcontest.c [hunk 3]
```diff
@@ -26,13 +26,65 @@
...
+#include "bands.h"
 #include "focm.h"
 #include "getctydata.h"
 #include "globalvars.h"
 #include "setcontest.h"
 #include "score.h"
 #include "tlf.h"
 
+/* contests where multiplier get determined from comment field can not be
+ shown in bandmap works also for modes with no multiplier at all */
```
Comment (zcsahok) [inline src/setcontest.c:39]: missing comma is sentence?

**Review by zcsahok — COMMENTED** [review]
:+1:
 
Not directly related:
- I had trouble understanding the code as `inxes[]` holds not indices as the name would suggest, but band mask values. Could we rename it to say `bandmask[]`?
- Do I see correctly that in `addmult()` the individual cases (if's) are mutually exclusive, just as in `score()`?

**Comment by dl1jbe** [comment]
> +1
> 
> Not directly related:
> 
>     * I had trouble understanding the code as `inxes[]` holds not indices as the name would suggest, but band mask values. Could we rename it to say `bandmask[]`?

Yes. That name has gone a long way. Let it finally named right.
> 
>     * Do I see correctly that in `addmult()` the individual cases (if's) are mutually exclusive, just as in `score()`?

I hope so (but are not 100% sure yet). If we can guarantee mutual exclusion the code can be simplified much. 
Btw, one of the main reasons for the switch to contest configs was the stronger  guarantee of mutual exclusion between the different point and multi scoring rules.

One downside of the actual implementation in these PR is that we now have one additional place where the logic for multiplier handling is located. I will try to bring it to a common code base to make sure we have the same eval rules for each case later.

File: src/setcontest.c (hunk 2 above)
Comment (dl1jbe) [reply src/setcontest.c]: Good catch, thanks

File: src/bandmap.c (hunk 1 above)
Comment (dl1jbe) [reply src/bandmap.c:450]: Good idea.

File: src/setcontest.c (hunk 3 above)
Comment (dl1jbe) [reply src/setcontest.c:39]: Did a rewording to (hopefully) make it more clear.

Comment (zcsahok) [reply src/setcontest.c]: Now clear, thanks.

**Comment by dl1jbe** [comment]
Finished handling of wpx style prefixes. Should be complete now.

* Last commit also fixed a segmentation fault if callmaster file could not be found.

**Comment by dl1jbe** [comment]
If all looks well, I would like to reorganize and squash the commits a little bit. Just give me a go if ok

**Review by zcsahok — APPROVED** [review]
Thanks!

**Comment by zcsahok** [comment]
I'm just thinking if it could be possible to structure multi handling as it is done for scoring: a contest would define its `get_multi()` functions and the core would just do the "bookkeeping".

**Comment by dl1jbe** [comment]
> I'm just thinking if it could be possible to structure multi handling as it is done for scoring: a contest would define its `get_multi()` functions and the core would just do the "bookkeeping".

That is one of the next goals. It should also be done for showscore() and get/checkexchange().

**Comment by dl1jbe** [comment]
Applied and pushed. Thanks for the feedback.

**Pull request closed without merging**